pkg/provider/azure_loadbalancer_backendpool_test.go

```diff
@@ -613,6 +713,21 @@ func TestGetBackendPrivateIPsNodeIPConfig(t *testing.T) {
 	assert.Equal(t, []string{"fe80::1"}, ipv6)
 }
 
+func TestGetBackendPrivateIPsNodeIP(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	lb := buildLBWithVMIPs(testClusterName, []string{"1.2.3.4", "fe80::1"})
+
+	az := GetTestCloud(ctrl)
+	az.VMSet = NewMockVMSet(ctrl)
+	bi := newBackendPoolTypeNodeIP(az)
+	svc := getTestService("svc1", "TCP", nil, false)
+	ipv4, ipv6 := bi.GetBackendPrivateIPs(testClusterName, &svc, lb)
```

Comment:
Also add a negative case?

Reply:
Updated. I added a case with no IPs and one with invalid IP. For the second one, I fixed `GetBackendPrivateIPs()` by updating. Are these what you expect?
```
if utilnet.IsIPv4String(*ipAddress) {
	backendPrivateIPv4s.Insert(*ipAddress)
} else {
	backendPrivateIPv6s.Insert(*ipAddress)
}
```
to
```
if utilnet.IsIPv4String(*ipAddress) {
	backendPrivateIPv4s.Insert(*ipAddress)
} else if utilnet.IsIPv6String(*ipAddress) {
	backendPrivateIPv6s.Insert(*ipAddress)
}
```